### ISSUE TITLE
Make Neuraxle compatible with Yellowbrick

### ISSUE THREAD

**Issue by guillaume-chevalier** [DistrictDataLabs/yellowbrick#1066]
**Describe the solution you'd like**
It would be cool to create a NeuraxleWrapper that would wrap any Neuraxle step or pipeline, just like how Neuraxle adapts scikit-learn steps into its core using [SKLearnWrapper](https://github.com/Neuraxio/Neuraxle/blob/1425d584817fa6b03ee303ae8815bb7e45418cb7/neuraxle/steps/sklearn.py#L39).

**Is your feature request related to a problem? Please describe.**
Related: 
- https://github.com/Neuraxio/Neuraxle/issues/340
- https://github.com/Neuraxio/Neuraxle/issues/342 (this suggested solution would avoid you to have to add Neuraxle to your requirements.txt if you do the same)

**Examples**
1. Neuraxle steps and pipeline don't directly inherit from sklearn steps, however they almost have the same API
2. Neuraxle steps and pipeline don't have a score method. Thus the NeuraxleWrapper could have such a method, being optionnally provided a metric to use at the construction of the wrapper or with a set_metric method on the wrapper. 

@alxkolm
